app/Dockerfile

```diff
@@ -1,14 +1,23 @@
-FROM ubuntu:14.04
+FROM debian:jessie
 
-RUN apt-get update && apt-get -y install curl netcat
-RUN mkdir -p /mattermost/data
+RUN apt-get update && apt-get -y install curl \
+ && rm -rf /var/lib/apt/lists/* \
+ && mkdir -p /opt/mattermost/data \
+ && useradd -r mattermost -U \
+ && cd /opt \
+ && curl https://releases.mattermost.com/3.0.2/mattermost-team-3.0.2-linux-amd64.tar.gz | tar -xvz
 
-RUN curl https://releases.mattermost.com/3.0.2/mattermost-team-3.0.2-linux-amd64.tar.gz | tar -xvz
+# grab gosu for easy step-down from root
+RUN gpg --keyserver pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4
+RUN curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.3/gosu-$(dpkg --print-architecture)" \
+ && curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.3/gosu-$(dpkg --print-architecture).asc" \
+ && gpg --verify /usr/local/bin/gosu.asc \
+ && rm /usr/local/bin/gosu.asc \
+ && chmod +x /usr/local/bin/gosu
 
 COPY config.template.json /
-COPY docker-entry.sh /
+COPY docker-entrypoint.sh /entrypoint.sh
 
-RUN chmod +x /docker-entry.sh
-ENTRYPOINT ["/docker-entry.sh"]
+ENTRYPOINT ["/entrypoint.sh"]
 
-EXPOSE 80
+EXPOSE 8080
```

Comment:
Obviously if you want to run anything with port 80, you need root.
But nginx drops to another user afterward:

```
root@364653b19ff2:/# ps -aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.0  0.0  31724   836 ?        Ss   Jun27   0:00 nginx:
master process nginx -g
www-data     5  0.0  0.0  32368  2104 ?        S    Jun27   3:46 nginx:
worker process
```

Again this morning I read about security in container recommending not
to run things as root inside (except if you use namespace shifting
capabilities)

Look, I'm fine, it is your repo after all. Just PR to the official repo
(with or without root) and you'll see what docker says.

I'm sorry, I have limited amount of time. I'm happy to share the work I
did on top of yours, but then feel free to use what you want/need.

On 24-08-2016 16:50, Yi EungJun wrote:
> In app/Dockerfile
> <https://github.com/mattermost/mattermost-docker/pull/31#discussion_r76083930>:
> 
>>  
>> -EXPOSE 80
>> +EXPOSE 8080
> 
> Could you let me know why we need to run it as nonroot to make this
> image official? In dockerhub, it seems that there are already a lot of
> official docker images that expose port 80 like nginx, httpd and
> wordpress. Exposing port 80 seems common.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mattermost/mattermost-docker/pull/31/files/cdf8330e9fb0b4d4197fe6a6436574c9d7d4c060#r76083930>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABxvHUKR0IPf_Y6jTHyZ0SDvNmE5EY6Rks5qjGg1gaJpZM4Iy56L>.
> 

-- 
I use PGP to protect our privacy, if you want to know more, you can
follow this
https://emailselfdefense.fsf.org/en/

If you have further questions, please do not hesitate to ask.
You can verify my public key here: https://keybase.io/pierreozoux